corehq/apps/hqwebapp/views.py

```diff
@@ -393,6 +398,43 @@ def logout(req):
 def retrieve_download(req, domain, download_id, template="hqwebapp/file_download.html"):
     return soil_views.retrieve_download(req, download_id, template)
 
+
+def dropbox_next_url(request, download_id):
+    return request.META.get('HTTP_REFERER', '/')
+
+
+@require_dropbox_session(next_url=dropbox_next_url)
+def dropbox_upload(request, download_id):
+    download = DownloadBase.get(download_id)
+    if download is None:
+        logging.error("Download file request for expired/nonexistent file requested")
+        raise Http404
+    else:
+        filename = download.get_filename()
```

Comment:
under the hood DownloadBase uses the NFS file system, so this should properly get the file. in the same vein, we shouldn't have to worry about clearing the file b/c that should already be taken care of, at least in the case of the multimedia download. the reason i don't think we should delete it straight away on the server is so that the user can still download the file without us having to regenerate it